chia/rpc/full_node_rpc_api.py

```diff
@@ -836,6 +839,39 @@ async def get_mempool_items_by_coin_name(self, request: dict[str, Any]) -> Endpo
 
         return {"mempool_items": [item.to_json_dict() for item in items]}
 
+    async def create_block_generator(self, _: dict[str, Any]) -> EndpointResult:
+        mempool_generator = BlockGenerator()
+        signature = G2Element()
+
+        # Grab best transactions from Mempool for given tip target
+        async with self.service.blockchain.priority_mutex.acquire(priority=BlockchainMutexPriority.low):
+            peak: Optional[BlockRecord] = self.service.blockchain.get_peak()
+
+            if peak is None:
+                return {
+                    "generator": mempool_generator.program,
+                    "refs": mempool_generator.generator_refs,
+                    "sig": signature,
+                }
+
+            # Finds the last transaction block before this one
+            curr_l_tb: BlockRecord = peak
+            while not curr_l_tb.is_transaction_block:
+                curr_l_tb = self.service.blockchain.block_record(curr_l_tb.prev_hash)
+
+            self.service.log.info("Beginning simulated block construction from mempool")
+            start_time = time.monotonic()
+
+            try:
+                result = await self.service.mempool_manager.create_block_generator(curr_l_tb.header_hash)
+                assert result is not None
+                mempool_generator, signature, _additions, _removals = result
```

Comment:
This will conflict/need to adapt to Arvid's new block generator refactor.

Reply:
yeah, I'll wait and rebase on top of this